app/utils/channel/index.ts

```diff
@@ -26,10 +55,21 @@ export function selectDefaultChannelForTeam(channels: Channel[], memberships: Ch
     return channel;
 }
 
-export function sortChannelsByDisplayName(locale: string, a: Channel, b: Channel): number {
+export function sortChannelsByDisplayName<T extends Channel|ChannelModel>(locale: string, a: T, b: T): number {
+    // if both channels have the display_name defined
+    const aDisplayName = 'display_name' in a ? a.display_name : a.displayName;
+    const bDisplayName = 'display_name' in b ? b.display_name : b.displayName;
```

Comment:
It is to deal with typescript typing. It may be a `Channel` type (`display_name`) or `ChannelModel` type (`displayName`). Therefore the ternary evaluation (if it is a Channel get `display_name`, if not, `displayName`).

If this was pure javascript we can do something like `aDisplayName = a.display_name || a.displayName`, but to deal with typescript types we go this direction.

Reply:
So a shorter version would be `a?.display_name ?? a.displayName;`

Reply:
On pure javascript, yes, that would be possible, but with the type constraints of TypeScript we need to be a bit more verbose.

Reply:
this is awesome!